app/helpers/utils.py

```diff
@@ -36,3 +45,58 @@ def get_logging_cfg():
 def init_logging():
     config = get_logging_cfg()
     logging.config.dictConfig(config)
+
+
+def prevent_erroneous_kml(kml_string):
+    # remove all attributes with on prefix and all script elements
+    kml_string = re.sub(r'on\w*=(".+?"|\'.+?\')', '', kml_string, flags=re.IGNORECASE)
+    kml_string = re.sub(
+        r'(<|&lt;)\s*\bscript\b.*?(>|&gt;).*?(<|&lt;)/\s*\bscript\b\s*(>|&gt;)',
+        ' ',
+        kml_string,
+        flags=re.IGNORECASE | re.DOTALL
+    )
+    return kml_string
+
+
+def bytes_to_mb(bytes, to, b_size=1024):
+    a = {'k': 1, 'm': 2, 'g': 3, 't': 4, 'p': 5, 'e': 6}
+    r = float(bytes)
+    return bytes / (b_size**a[to])
+
+
+def validate_content_type(content):
+
+    def inner_decorator(func):
+
+        @wraps(func)
+        def wrapped(*args, **kwargs):
+            if request.content_type != content:
+                abort(make_error_msg(415, 'Unsupported Media Type'))
+            return func(*args, **kwargs)
+
+        return wrapped
+
+    return inner_decorator
+
+
+def validate_kml_string(kml_string):
+
+    max_file_size = 1024 * 1024 * 2
+    file_size_in_mb = bytes_to_mb(max_file_size, 'm')
+    kml_string_mb = bytes_to_mb(len(kml_string), 'm')
+
+    if len(kml_string) > max_file_size:
+        error_msg = 'File size exceed {} MB. The actual file size is {} MB'.format(
+            file_size_in_mb, kml_string_mb
+        )
+        logger.error(error_msg)
+        abort(413, error_msg)
```

Comment:
You don't need to generate the mb version, just do it on the fly in the error message formatting

`f"File size exceed {bytes_to(max_file_size, 'MB')} MB. The actual file size is {bytes_to(len(kml_string), 'MB')} MB"`
or 
`'File size exceed {} MB. The actual file size is {} MB'.format(
            bytes_to(max_file_size, 'MB'), bytes_to(len(kml_string), 'MB')
        )`